### PR TITLE
Add instanceIdIncludeHostname config property

### PR DESCRIPTION
**Pull request by tianzhipeng-git**
Add instanceIdIncludeHostname config property to control 'whether hostname is included into the default instance id when registering service' #487

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@tianzhipeng-git Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-consul&pullRequestId=492)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-consul&pullRequestId=492) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@tianzhipeng-git Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-consul&pullRequestId=492)!

File: spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryProperties.java [hunk 1]
```diff
@@ -115,6 +115,11 @@
 	/** Service instance group. */
 	private String instanceGroup;
 
+	/**
+	 * Whether hostname is included into the default instance id when registering service.
+	 */
+	private boolean instanceIdIncludeHostname;
```
Comment (spencergibb) [inline spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryProperties.java]: Can you change it to `includeHostnameInInstanceId`?

File: spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/serviceregistry/ConsulAutoRegistration.java [hunk 2]
```diff
@@ -166,10 +166,10 @@ public static void managementCustomize(
 	public static String getInstanceId(ConsulDiscoveryProperties properties,
 			ApplicationContext context) {
 		if (!StringUtils.hasText(properties.getInstanceId())) {
-			return normalizeForDns(
-					IdUtils.getDefaultInstanceId(context.getEnvironment(), false));
+			return IdUtils.getDefaultInstanceId(context.getEnvironment(),
```
Comment (spencergibb) [inline spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/serviceregistry/ConsulAutoRegistration.java]: You removed `normalizeForDns()`, please add it back and undo changes to tests.

File: spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryProperties.java (hunk 1 above)
Comment (tianzhipeng-git) [reply spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryProperties.java]: Of course.

File: spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/serviceregistry/ConsulAutoRegistration.java (hunk 2 above)
Comment (tianzhipeng-git) [reply spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/serviceregistry/ConsulAutoRegistration.java]: Oh, I removed `normalizeForDns()` intentionally. As comments in #487 and #331 , maybe this method should be rewrite. And consul don't provide dns query for InstanceId, so i think it is no need to `normalizeForDns` for instanceId.

Comment (tianzhipeng-git) [reply spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/serviceregistry/ConsulAutoRegistration.java]: I reset some commits and get it back. Maybe improve later.

**Comment by spencergibb** [comment]
Can you rebase and retarget the 2.1.x branch?

**Comment by tianzhipeng-git** [comment]
Maybe I can cherrypick these little changes to another branch and open another pr, if you need.

**Comment by tianzhipeng-git** [comment]
Or just close this pr, not an essential change actually.

**Pull request closed without merging**